code/game/machinery/doors/firedoor.dm

```diff
@@ -25,7 +26,6 @@
 	closingLayer = CLOSED_FIREDOOR_LAYER
 	assemblytype = /obj/structure/firelock_frame
 	armor = list(melee = 30, bullet = 30, laser = 20, energy = 20, bomb = 10, bio = 100, rad = 100, fire = 95, acid = 70)
-	CanAtmosPass = ATMOS_PASS_PROC

```

Comment:
Because it'd return YES, allowing atmos to pass; only the directional/border firedoor should have had this.
firedoors inherit ATMOS_PASS_DENSITY from doors otherwise.